test/internal_rules.jl

```diff
@@ -136,282 +136,66 @@ end
+                    _A = collect(exp(TS(rand(Te, 4, 4))))
+                    A = TS(_A, uplo)
+                    are_activities_compatible(Tret, TA) || continue
+                    test_forward(cholesky, Tret, (A, TA))
+                    test_reverse(cholesky, Tret, (A, TA))
+                end
+            end
         end
-        return J
-    end
-
-    function JdxdA(driver, A, b)
-        db = zeros(length(b))
-        J = zeros(length(b), length(b))
-        for i in 1:length(b)
-            db[i] = 1.0
-            dx = A\db
-            db[i] = 0.0
-            J[i, :] = dx
+        @testset "without wrapper arguments" begin
+            _square(A) = A * A'
+            @testset for Te in (Float64,)
+                @testset for TA in (Const, Duplicated), Tret in (Const, Duplicated)
+                    A = rand(Te, 4, 4)
+                    are_activities_compatible(Tret, TA) || continue
+                    test_forward(cholesky ∘ _square, Tret, (A, TA))
+                    test_reverse(cholesky ∘ _square, Tret, (A, TA))
+                end
+            end
         end
-        return J
     end
-    
-    @testset "Testing $op" for (op, driver, driver_NC) in (
-        (:\, divdriver, divdriver_NC),
-        (:\, divdriver_herm, divdriver_NC),
-        (:\, divdriver_sym, divdriver_NC),
-        (:ldiv!, ldivdriver, ldivdriver_NC),
-        (:ldiv!, ldivdriver_herm, ldivdriver_NC),
-        (:ldiv!, ldivdriver_sym, ldivdriver_NC)
-    )
-        A, b = symmetric_definite(10)
-        n = length(b)
-        A = Matrix(A)
-        x = zeros(n)
-        x = driver(x, A, b)
-        fdm = forward_fdm(2, 1);
-
-        function b_one(b)
-            _x = zeros(length(b))
-            driver(_x,A,b)
-            return _x
-        end
 
-        fdJ = op==:\ ? FiniteDifferences.jacobian(fdm, b_one, copy(b))[1] : nothing
-        fwdJ = fwdJdxdb(driver, A, b)
-        revJ = revJdxdb(driver, A, b)
-        batchedrevJ = batchedrevJdxdb(driver, A, b)
-        batchedfwdJ = batchedfwdJdxdb(driver, A, b)
-        J = Jdxdb(driver, A, b)
-
-        if op == :\
-            @test isapprox(fwdJ, fdJ)
-        end
-
-        @test isapprox(fwdJ, revJ)
-        @test isapprox(fwdJ, batchedrevJ)
-        @test isapprox(fwdJ, batchedfwdJ)
-
-        fwdJ = const_fwdJdxdb(driver_NC, cholesky(A), b)
-        revJ = const_revJdxdb(driver_NC, cholesky(A), b)
-        if op == :\
-            @test isapprox(fwdJ, fdJ)
-        end
-        @test isapprox(fwdJ, revJ)
-
-        function h(A, b)
-            C = cholesky(A)
-            b2 = copy(b)
-            ldiv!(C, b2)
-            @inbounds b2[1]
+    @testset "Linear solve for `Cholesky`" begin
+        activities = (Const, Duplicated, DuplicatedNoNeed, BatchDuplicated,
+                      BatchDuplicatedNoNeed)
+        @testset for Te in (Float64, ComplexF64), uplo in ('L', 'U')
+            C = Cholesky(I + rand(Te, 4, 4), uplo, 0)
```

Comment:
Could you add a brief not why the addition of `I` is necessary?

Reply:
Done!